neps/nep-0452.md

```diff
@@ -0,0 +1,213 @@
+/// * The assets *MUST* be sent to the `new_account_id` passed in.
+/// * The newly created account *MUST* have a new access key added to its account (either       
+///   full or limited access) in the same receipt that the account was created in. 
+/// 
+/// Arguments
+/// * `new_account_id`: the valid NEAR account which is being created and should 
+///   receive the linkdrop assets
+/// * `new_public_key`: the valid public key that should be used for the access key added to         
+/// The Public key must be in a binary format with base58 string serialization with human-readable curve. The key types currently supported are secp256k1 and ed25519. Ed25519 public keys accepted are 32 bytes and secp256k1 keys are the uncompressed 64 format.
+///   the newly created account.
+function create_account_and_claim(new_account_id: string, new_public_key: string) -> Promise
+```
+
+## Example Scenarios
+
+*Pre-requisite Steps*: Linkdrop creation:
+The linkdrop creator that has an account and some $NEAR:
+
+- creates a keypair locally (`pubKey1`, `privKey1`). (The keypair is not written to chain at this time)
+- calls send on the contract and passes in the pubKey1 as an argument as well as the desired balance for the linkdrop.
+- The contract will map the `pubKey1` to the desired balance for the linkdrop.
+- The contract will then add the `pubKey1` as a function call access key with the ability to call claim and create_account_and_claim. This means that anyone with the `privKey1` that was created locally, can claim this linkdrop.
+- Linkdrop creator will then create a link to send to someone that contains this privKey1. The link may follow the format: wallet.testnet.near.org/linkdrop/{fundingContractAccountId}/{linkdropKeyPairSecretKey}?redirectUrl={redirectUrl}
+`fundingContractAccountId`: The contract accountId that was used to send the funds.
+`linkdropKeyPairSecretKey`: The corresponding secret key to the public key sent to the contract.
+`redirectUrl`: The url that wallet will redirect to after funds are successfully claimed to an existing account. The URL is sent the accountId used to claim the funds as a query param.
+
+### Claiming a linkdrop without a NEAR Account
+
+The receiver of the link that is claiming the linkdrop:
+
+- Receives the link which redirects them to a linkdrop claiming page (such as NEAR wallet).
+- The linkdrop claim page calls `get_key_information` to display information associated with the linkdrop
+- The linkdrop claiming page creates a new keypair (`pubKey2`, `privKey2`) locally. (The keypair is not written to chain at this time)
+- Receiver will then choose an account ID such as alice.near.
+- Wallet will then use the privKey1 which has access to call `claim` and `create_account_and_claim` in order to call `create_account_and_claim` on the contract.
+- It will pass in `pubKey2` which will be used to create a full access key for the new account.
```

Comment:
There is a lot of implicitness in "It will pass in `pubKey2`". Please, make it more specific - talk about forming function call arguments for `claim` and then how they are different from `create_account_and_claim`.